crates/sui-indexer-alt/src/models/objects.rs

```diff
@@ -11,3 +17,40 @@ pub struct StoredObject {
     pub object_version: i64,
     pub serialized_object: Option<Vec<u8>>,
 }
+
+#[derive(AsExpression, Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+#[diesel(sql_type = SmallInt)]
+#[repr(i16)]
+pub enum StoredOwnerKind {
+    Immutable = 0,
+    Address = 1,
+    Object = 2,
+    Shared = 3,
+}
+
+#[derive(Insertable, Debug, Clone)]
+#[diesel(table_name = sum_obj_types, primary_key(object_id, object_version))]
+pub struct StoredSumObjType {
+    pub object_id: Vec<u8>,
+    pub object_version: i64,
+    pub owner_kind: StoredOwnerKind,
```

Comment:
@emmazzz, In the tx indices PR, I asked whether it was possible to use an `enum` directly in a model type like this -- it is possible! You need to have it `derive(AsExpression)` and specify the `diesel(sql_type = ...)` for it, and then implement the `ToSql` for it.

It's a bit boiler-plate-y, but then you don't need a conversion on read/write, which is nice.

Reply:
I think it's only the read case we need to worry about, but it's a good question, I don't know.

Reply:
Okay, so I now have the answer to this: As written, you won't be able to select `StoredSumObjType` because `StoredOwnerKind` does not implement `FromSqlRow` or `FromSql`.

That seems not ideal, these model types will be used on the read side as well. Once I added those impls (I'll update the PR), I was able to add an error case for that too, and then all is well.